include/libdnf/system/state.hpp

```diff
@@ -30,6 +32,41 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 namespace libdnf::system {
 
+class PackageState {
+public:
+    bool userinstalled{false};
+};
+
+class NevraState {
+public:
+    std::string from_repo;
```

Comment:
For the NevraState state it quite easy to determine not initialized or a default value because the empty string is not valid repoid. Again it is not a blocker but somehow I feel it is not right.

Reply:
The class is not on the interface as of now, so the user can't make this mistake. It's just a dataholder. On the interface it's `void set_package_from_repo(const std::string & nevra, const std::string & from_repo);`. Not that it prevents passing an empty string... But it doesn't have the "uninitialized class" issue.

Again, this will need a bit of cleaning up later.